src/xunit.performance.api/ETWProfiler.cs

```diff
@@ -107,9 +119,196 @@ public static void Profile(string assemblyFileName, string sessionName, Action a
+                    newRow[col2_iterations] = count.ToString();
+                    newRow[col3_average] = avg.ToString();
+                    newRow[col4_stdevs] = stdev_s.ToString();
+                    newRow[col5_min] = min.ToString();
+                    newRow[col6_max] = max.ToString();
+                }
+            }
+
+            return dt;
+        }
+
+        private static AssemblyModel GetAssemblyModel(
+            string assemblyFileName,
+            string etlFileName,
+            string sessionName,
+            IEnumerable<PerformanceTestMessage> performanceTestMessages)
+        {
+            using (var reader = GetEtwReader(etlFileName, sessionName, performanceTestMessages))
+            {
+                var assemblyModel = new AssemblyModel
+                {
+                    Name = Path.GetFileName(assemblyFileName),
+                    Collection = new List<TestModel>()
+                };
+
+                foreach (var test in performanceTestMessages)
+                {
+                    var metrics = new List<MetricModel>();
+                    foreach (var metric in test.Metrics)
+                    {
+                        metrics.Add(new MetricModel
+                        {
+                            DisplayName = metric.DisplayName,
+                            Name = metric.Id,
+                            Unit = metric.Unit,
+                        });
+                    }
+
+                    var testModel = new TestModel
+                    {
+                        Name = test.TestCase.DisplayName,
+                        Method = test.TestCase.TestMethod.Method.Name,
+                        ClassName = test.TestCase.TestMethod.TestClass.Class.Name,
+                        Performance = new PerformanceModel { Metrics = metrics, IterationModels = new List<IterationModel>() },
+                    };
+
+                    var values = reader.GetValues(testModel.Name);
+                    foreach (var dict in values)
+                    {
+                        var iterationModel = new IterationModel { Iteration = new Dictionary<string, double>() };
+
+                        foreach (var kvp in dict)
+                            iterationModel.Iteration.Add(kvp.Key, kvp.Value);
+
+                        if (iterationModel.Iteration.Count > 0)
+                            testModel.Performance.IterationModels.Add(iterationModel);
+                    }
+
+                    assemblyModel.Collection.Add(testModel);
+                }
+
+                return assemblyModel;
+            }
+        }
+
+        private static EtwPerformanceMetricEvaluationContext GetEtwReader(
+            string fileName,
+            string sessionName,
+            IEnumerable<PerformanceTestMessage> performanceTestMessages)
+        {
+            using (var source = new ETWTraceEventSource(fileName))
+            {
+                if (source.EventsLost > 0)
```

Comment:
Gotcha, thanks. Per our offline conversation, I think this is good to keep as-is for now.

---
In reply to: [101368446](https://github.com/Microsoft/xunit-performance/pull/191#discussion_r101368446) [](ancestors = 101368446,101316411)